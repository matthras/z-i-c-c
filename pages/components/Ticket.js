const Ticket = props => (
  <ul>
    <li>ID: {props.id}</li>
    <li>Subject: {props.subject}</li>
  </ul>
)

export default Ticket;