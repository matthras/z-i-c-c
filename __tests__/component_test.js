import React from 'react'
import { render } from 'react-testing-library'

// Tests that each component renders as expected with pre-filled information

// Test for Ticket - given a single dummy ticket data, renders the component as expected.
import Ticket from '../pages/components/Ticket'

describe('Ticket snapshot with example ticket data', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Ticket
        id="36"
        status="open"
        priority="high"
        subject="this is a test ticket"
        description="of course this is fake"
        created_at=""
        recipient="you!"
        url="dummy.url" 
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})


// Test for TicketList - given a list of dummy ticket data, correctly calculates pagination factors (e.g. correct number of tickets, and correct number of pages)