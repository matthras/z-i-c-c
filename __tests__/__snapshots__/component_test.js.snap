// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket snapshot with example ticket data should match snapshot 1`] = `
<DocumentFragment>
  <div
    id="36"
  >
    <h2>
      ID: 36 | Status: open | Priority: high
    </h2>
    <h3>
      Subject: this is a test ticket
    </h3>
    <h4>
      Description: of course this is fake
    </h4>
    <ul>
      <li>
        Created: 
      </li>
      <li>
        Recipient: you!
      </li>
      <li>
        URL: dummy.url
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
